### DISCUSSION TITLE
Is there a rule to disallow else branches in when statements?

### DISCUSSION THREAD

**Discussion by G00fY2** [detekt/detekt#4619] [Q&A] [answered]
Is there already a rule that allows you to report else branches in when statements for exhaustive conditions (enums, sealed classes)?

I am looking for a way to prevent such behaviour.
```kotlin
enum class MyEnum {
    ENUM1, ENUM2, ENUM3, ENUM4
}
```
### Noncompliant Code:
```kotlin
when (type) {
    ENUM1 -> // ok
    ENUM2 -> // ok
    else ->  // detekt should report this else
}
```
### Compliant Code:
```kotlin
when (type) {
    ENUM1 -> // ok
    ENUM2 -> // ok
    ENUM3 -> // ok
    ENUM4 -> // ok
}
```

Maybe I am missing a rule that does that?

**Comment by cortinico** [comment, marked as answer]
Yes, it's the `RedundantElseInWhen` rule :)

**Reply by G00fY2** [reply]
@cortinico Thanks for answering. Maybe my example code was missing the enum class to make sure what I meant (I edited my initial post).

I think `RedundantElseInWhen` is only there to find obsolete else branches.

I also tried `MissingWhenCase` in conjunction with `allowElseExpression: false`. But this also does not report my example above.

Looks like there is no such rule by now?

**Reply by cortinico** [reply]
That's correct. Thanks for updating the post. There is no such rule and could be created.

**Reply by G00fY2** [reply]
Thanks, then my question is answered. Maybe I find time to contribute such a rule. :)

**Comment by G00fY2** [comment]
I actually added this "missing" rule with #4632
